woocommerce/woocommerce-android
# issue/3895-edittext-clears-itself

Fixes #3895 - Right now when a `WCMaterialOutlinedEditTextView` receives `IME_ACTION_DONE`, it clears the editor's text. This logic was added so the text would be cleared when adding a tag to a product, leaving a blank editor for the next tag to be entered.

The problem with this is that if you try to read the editor text after `IME_ACTION_DONE`, rather than returning the user's entry it returns an empty string because the editor's text has been cleared.

This PR resolves this by:

1. Not clearing the editor text after `IME_ACTION_DONE`
2. Clearing the text in the tag editor after `IME_ACTION_DONE`

To test:

- Go to product detail for a product variation
- Tap on Variations to open the variation list
- Select "Edit attributes" from the options menu
- Tap a local attribute (one specific to that product, not a global attribute)
- Select Rename from the options menu
- Rename the attribute
- Press the Done button on the keyboard
- Note the attribute gets correctly renamed

And then:

- Go to product detail for any product type
- Edit the product's tags
- Enter a new tag
- Note that tapping Done on the keyboard adds the new tag and clears the existing entry

Update release notes:
